composer/lib/dependabot/composer/helpers.rb

```diff
@@ -7,21 +7,29 @@ module Composer
     module Helpers
       # From composers json-schema: https://getcomposer.org/schema.json
       COMPOSER_V2_NAME_REGEX = %r{^[a-z0-9]([_.-]?[a-z0-9]+)*/[a-z0-9](([_.]?|-{0,2})[a-z0-9]+)*$}.freeze
+      # From https://github.com/composer/composer/blob/b7d770659b4e3ef21423bd67ade935572913a4c1/src/Composer/Repository/PlatformRepository.php#L33
+      PLATFORM_PACKAGE_REGEX = /
+        ^(?:php(?:-64bit|-ipv6|-zts|-debug)?|hhvm|(?:ext|lib)-[a-z0-9](?:[_.-]?[a-z0-9]+)*
+        |composer-(?:plugin|runtime)-api)$
+      /x.freeze
 
       def self.composer_version(composer_json, parsed_lockfile = nil)
-        return "v1" if composer_json["name"] && composer_json["name"] !~ COMPOSER_V2_NAME_REGEX
-        return "v1" if invalid_v2_requirement?(composer_json)
-        return "v2" unless parsed_lockfile && parsed_lockfile["plugin-api-version"]
+        if parsed_lockfile && parsed_lockfile["plugin-api-version"]
+          version = Composer::Version.new(parsed_lockfile["plugin-api-version"])
+          return version.canonical_segments.first == 1 ? "v1" : "v2"
+        else
+          return "v1" if composer_json["name"] && composer_json["name"] !~ COMPOSER_V2_NAME_REGEX
+          return "v1" if invalid_v2_requirement?(composer_json)
+        end
 
-        version = Composer::Version.new(parsed_lockfile["plugin-api-version"])
-        version.canonical_segments.first == 1 ? "v1" : "v2"
+        "v2"
       end
 
       def self.invalid_v2_requirement?(composer_json)
```

Comment:
Thoughts on adding a comment here to clarify the rules?

```suggestion
      # NOTE: Composer v2 validates `requires` keys to have the format `vendor/package-name` but
      # allows certain "platform" specific keys, starting with `php-`, `ext-` or `lib-`
      def self.invalid_v2_requirement?(composer_json)
```